src/objects.c

```diff
@@ -236,22 +243,39 @@ static void obj_rm_from_pool(P11PROV_OBJ *obj)
     }
 
     /* LOCKED SECTION ------------- */
-    if (obj->poolid >= pool->size || pool->objects[obj->poolid] != obj) {
+    int idx = obj->poolid;
+    if (idx >= pool->size || pool->objects[idx] != obj) {
         ret = CKR_GENERAL_ERROR;
-        P11PROV_raise(pool->provctx, ret, "Objects pool in inconsistent state");
+        if (obj->poolid >= pool->size) {
+            P11PROV_raise(pool->provctx, ret,
+                          "Objects pool in inconsistent state - small pool "
+                          "(idx=%d, size=%d)",
+                          idx, pool->size);
+        } else {
+            P11PROV_raise(pool->provctx, ret,
+                          "Objects pool in inconsistent state - obj already "
+                          "removed (idx=%d, pool_obj=%p, obj=%p)",
+                          idx, pool->objects[idx], obj);
+        }
```

Comment:
perhaps it is time to split the whole error from line 247, instead of inferring that the second banch here is caused by `pool->objects[idx] != obj)`.


IE do:

```
if (idx >= pool->size) {
  ret = ..
  P11PROV_raise(..
  goto done;
}
if (pool->objects[idx] != obj) {
  ret = ...
  P11PROV_raise(..
  goto done;
}
```

Reply:
yeah that looks better. Done